sdks/go/pkg/beam/core/runtime/exec/plan.go

```diff
@@ -38,7 +39,7 @@ type Plan struct {
 	bf          *bundleFinalizer
 	checkpoints []*Checkpoint
 
-	status Status
+	status Status // Uses atomic getter and setter to avoid dataraces on Splits.
```

Comment:
Splits are how Beam (and thus Dataflow) allow for Liquid Sharding. That is, if one bundle is taking a long time, the runner can request the SDK to give up some fraction of it's remaining work so it can be processed in parallel. (This would either be via channel or sub-element splitting with SDFs, but that's a different topic.)